text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,703 @@
+
+The limit of the address range occupied by this mapping.
+
+##### Field `file_offset`
+
+Offset in the binary that corresponds to the first mapped address.
+
+##### Field `filename_index`
+
+The object this entry is loaded from.  This can be a filename on
+disk for the main binary and shared libraries, or virtual
+abstractions like "[vdso]". Index into string table
+
+##### Field `build_id_index`
+
+A string that uniquely identifies a particular program version
+with high probability. E.g., for binaries generated by GNU tools,
+it could be the contents of the .note.gnu.build-id field. Index into string table
+
+##### Field `attribute_set_indices`
+
+Reference to an attribute set from the Profile's attribute set table.
+</details>
+
+#### Message `Function`
+
+Represents a function in a source file. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `name_index`
+
+Name of the function, in human-readable form if available. Index into string table
+
+##### Field `system_name_index`
+
+Name of the function, as identified by the system.
+For instance, it can be a C++ mangled name. Index into string table
+
+##### Field `filename_index`
+
+Source file containing the function. Index into string table
+
+##### Field `start_line`
+
+Line number in source file.
+</details>
+
+<!-- messages -->
+
+### Example Payloads
+
+#### Simple example
+
+Considering the following example presented in a modified collapsed format:
+
+```
+foo;bar;baz 100 region=us,trace_id=0x01020304010203040102030401020304,span_id=0x9999999999999999 1687841528000000
+foo;bar 200 region=us
+```
+
+It represents 2 samples:
+
+* one for stacktrace `foo;bar;baz` with value `100`, attributes `region=us`, linked to trace `0x01020304010203040102030401020304` and span `0x9999999999999999`, and timestamp `1687841528000000`
+* one for stacktrace `foo;bar` with value `200`, attributes `region=us`, no link, no timestamp
+
+The resulting profile in OTLP format would look like this (in YAML format):
+
+```
+resource_profiles:
+  - resource:
+      attributes: null
+    schema_url: todo
+    scope_profiles:
+      - profiles:
+          - attribute_sets:
+              - attributes: null
+              - attributes:
+                  - key: region
+                    value:
+                      Value:
+                        string_value: us
+            functions:
+              - name_index: 1
+              - name_index: 2
+              - name_index: 3
+            links:
+              - span_id: ""
+                trace_id: ""
+              - span_id: "9999999999999999"
+                trace_id: "01020304010203040102030401020304"
+            locations:
+              - line:
+                  - {}
+              - line:
+                  - function_index: 1
+              - line:
+                  - function_index: 2
+            profile_types:
```

Comment:
I understand this is a simplified example focused on showing how data is linked, however, adding at least `type_index` and `unit_index` (which I think aren't optional) would make it clearer to me. Without those, it's not clear what is measured.

Reply:
Update the OTEP